lib/server/backends/static-sites.ts

```diff
@@ -1,47 +1,114 @@
+import fs from "fs"
+import path from "path"
+import { cwd } from "process"
+
 import env from "../environment"
 import { ModelsBackend, ValidatedRequest } from "../server-types"
 import { Site } from "../../models"
 import { DEFAULT_SITE } from "../../defaults"
-import { DEMO_SITE } from "../../sites"
+import { TimerLike } from "../../types"
+
+const asyncfs = fs.promises
 
 
 export type JsonSiteBackendOptions = {
 
   /** Base path to load sites from */
   basePath: string,
 
-  /** Minimum time in ms to wait before reloading a site from disk */
+  /**
+   * Minimum time in ms to wait before reloading a site from disk.
+   * Set to -1 for no reloading.
+   */
   refreshInterval: number,
-
-  /** Map of site names to local paths */
-  sites: {
-    [siteName: string]: string,
-  }
 }
 
 /**
  * Serves sites from named JSON definitions on the server
  */
 export function getStaticSiteBackend(initOptions: Partial<JsonSiteBackendOptions> = {}): ModelsBackend {
   const options = Object.assign({}, DEFAULT_STATIC_SITE_BACKEND_OPTIONS, initOptions)
+  let sites: { [name: string]: Site } = {}
+  let lastScan = 0
+
+  function logError(err: Error) {
+    console.warn(`[IB] Error scanning sites: ${err}`)
+  }
+
   return {
+    async connect() {
+      try {
+        sites = await scanSites(env.IB_STATIC_SITES_PATH, sites, lastScan)
+      } catch (err) {
+        logError(err)
+      }
+    },
+
     async getSiteConfig(req: ValidatedRequest): Promise<Site> {
+      const { siteName } = req
+      const { refreshInterval } = options
+      const now = new Date().getTime()
 
-      // STUB
-      if (req.siteName === "demo") {
-        return DEMO_SITE
-      } else {
+      if (refreshInterval !== -1 && (now - lastScan > refreshInterval)) {
```

Comment:
this is a lot to unpack in one line. can you split into several booleans that describe what's going?